lib/t_web/controllers/media_controller.ex

```diff
@@ -1,18 +1,26 @@
 defmodule TWeb.MediaController do
   use TWeb, :controller
-  alias T.Accounts
+  alias T.{Accounts, Voicemail}
 
   def create_upload_form(conn, %{"media" => params}) do
-    "image/" <> _rest =
-      content_type =
+    content_type =
       case params do
         %{"content-type" => content_type} -> content_type
         %{"extension" => extension} -> MIME.type(extension)
       end
 
+    json(conn, upload_form(content_type))
+  end
+
+  # if it's an image -> it's for profile story photo
+  defp upload_form("image/" <> _rest = content_type) do
     {:ok, %{"key" => key} = fields} = Accounts.photo_upload_form(content_type)
-    url = Accounts.photo_s3_url()
+    %{url: Accounts.photo_s3_url(), key: key, fields: fields}
+  end
 
-    json(conn, %{url: url, key: key, fields: fields})
+  # if it's an audio/aac -> it's for voicemail
+  defp upload_form("audio/aac" = content_type) do
```

Comment:
any rationale for choosing `aac` specifically?

Reply:
does apple support anything else?

Reply:
https://stackoverflow.com/questions/6867994/what-are-the-formats-supported-in-avaudiorecorder-for-recording-sound
seems to support some, yeah

Reply:
for recording ...